client/context/query.go

```diff
@@ -29,28 +29,34 @@ func (ctx CLIContext) GetNode() (rpcclient.Client, error) {
 	return ctx.Client, nil
 }
 
-// Query performs a query for information about the connected node.
-func (ctx CLIContext) Query(path string, data cmn.HexBytes) (res []byte, err error) {
+// Query performs a query to a Tendermint node with the provided path.
+// It returns the result and height of the query upon success
+// or an error if the query fails.
+func (ctx CLIContext) Query(path string, data cmn.HexBytes) ([]byte, int64, error) {
 	return ctx.query(path, data)
 }
 
-// Query information about the connected node with a data payload
-func (ctx CLIContext) QueryWithData(path string, data []byte) (res []byte, err error) {
+// QueryWithData performs a query to a Tendermint node with the provided path
+// and a data payload. It returns the result and height of the query upon success
+// or an error if the query fails.
+func (ctx CLIContext) QueryWithData(path string, data []byte) ([]byte, int64, error) {
```

Comment:
Hmmm, looks like `QueryWithData` is identical to `Query`! How about we simply remove `QueryWithData`?